tonic/transforms.py

```diff
@@ -316,6 +316,62 @@ def __call__(self, events):
             return functional.drop_event_numpy(events, ratio)
 
 
+@dataclass(frozen=True)
+class EventDownsampling:
+    """Applies EventDownsampling from the paper "Insect-inspired Spatio-temporal Downsampling of Event-based
+        Input."
+        Allows:
+            1. Creating time bins to round time values to a certain value.
+            2a. Tonic's normal "naive" topographical scaling method to perform spatio-temporal event-based downsampling
+            2b. Integrator based method to perform spatio-temporal event-based downsampling
+            2c. Differentiator based method to perform spatio-temporal event-based downsampling
+            3. Feedback inhibition circuit.
+            
+    Parameters:
+        sensor_size (Tuple): size of the sensor that was used [W,H,P]
+        target_size (Tuple): size of the desired resolution [W,H]
+        
+    Example:
+        >>> transform = tonic.transforms.EventDownsampling(sensor_size=(640,480,2), target_size=(20,15))
+    """
+    
+    sensor_size: Tuple[int, int, int]
+    target_size: Tuple[int, int]
+    
+    def __call__(self, events):
+        downsampling_method = np.random.choice(['naive', 'integrator', 'differentiator'], 1)[0]
```

Comment:
why is this chosen randomly?

Reply:
Added another parameter for the downsampling_method